src/rail/estimation/algos/deepdisc.py

```diff
@@ -54,12 +55,22 @@ def train(config, training_metadata, train_head=True):
 
     val_per = epoch
 
+    # Create slices for the input data
+    total_images = len(all_metadata)
+    train_slice = slice( int(np.floor(total_images * config[training_percent])) )
+    eval_slice = slice( int(np.floor(total_images * config[training_percent])) + 1, total_images )
+
     mapper = RedshiftDictMapper(
         DC2ImageReader(), lambda dataset_dict: dataset_dict["filename"]
     ).map_data
 
-    loader = d2data.build_detection_train_loader(
-        training_metadata, mapper=mapper, total_batch_size=batch_size
+    training_loader = d2data.build_detection_train_loader(
+        all_metadata[train_slice], mapper=mapper, total_batch_size=batch_size
+    )
+
+    #! Grant to confirm these input variables
```

Comment:
@grantmerz please confirm that this looks correct for `build_detection_test_loader`.

Reply:
the `total_batch_size` keyword in line 73 should be `batch_size`.  The keyword in line 68 should stay as is